sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs

```diff
@@ -1040,13 +1047,19 @@ public static partial class SpecializedBlobExtensions
         /// <returns>A new <see cref="AppendBlobClient"/> instance.</returns>
         public static AppendBlobClient GetAppendBlobClient(
             this BlobContainerClient client,
-            string blobName) =>
-            new AppendBlobClient(
+            string blobName)
+        {
+            if (client.ClientSideEncryption != default)
+            {
+                throw Errors.ClientSideEncryption.TypeNotSupported(typeof(BlockBlobClient));
```

Comment:
Should this be typeof(AppendBlobClient)? I think this was also missed in PageBlobClient